tests/test_reconcile.py

```diff
@@ -28,7 +28,7 @@ def test_deal_with_negation(self):
         df1 = deal_with_negation(self.msdf1.df)
         self.assertEqual(8, len(df1.index))
         df2 = deal_with_negation(self.msdf2.df)
-        self.assertEqual(12, len(df2.index))
+        self.assertEqual(14, len(df2.index))
```

Comment:
This is what concerns me most, can you figure out what these two lines are - the update should not have change the number of rows in the reconciled part.

Reply:
Yeah now I remember. ..... The negative - positive splitting was because we wanted to enforce the rule of "negative trumps positive" in case of a confidence tie. Because we are doing this, grouping by `predicate_modifier` makes no sense. Hence ~`KEY_FEATURES`~ => `TRIPLE_IDS`